tests/test_immutableset.py

```diff
@@ -195,6 +195,18 @@ def test_index(self):
         with self.assertRaises(ValueError):
             s.index("z")
 
+    def test_singleton_index(self):
+        s = ImmutableSet.of([1])
+
+        try:
+            s[0]
+        except IndexError as e:
```

Comment:
I don't think you need the `try ... except` - instead try...

```
self.assertEqual(1, s[0])
self.assertEqual(1, s[-1])
```

If an exception is thrown, the testing framework will catch it and fail the test.